rust/protocol/src/identity_key.rs

```diff
@@ -15,31 +21,42 @@ use prost::Message;
 const ALTERNATE_IDENTITY_SIGNATURE_PREFIX_1: &[u8] = &[0xFF; 32];
 const ALTERNATE_IDENTITY_SIGNATURE_PREFIX_2: &[u8] = b"Signal_PNI_Signature";
 
+/// A public key that represents the identity of a user.
+///
+/// Wrapper for [`PublicKey`].
 #[derive(Debug, PartialOrd, Ord, PartialEq, Eq, Clone, Copy)]
 pub struct IdentityKey {
     public_key: PublicKey,
 }
 
 impl IdentityKey {
+    /// Initialize a public-facing identity from a public key.
     pub fn new(public_key: PublicKey) -> Self {
         Self { public_key }
     }
 
+    /// Return the public key representing this identity.
     #[inline]
     pub fn public_key(&self) -> &PublicKey {
         &self.public_key
     }
 
+    /// Return an owned byte slice which can be deserialized with [`Self::decode`].
     #[inline]
     pub fn serialize(&self) -> Box<[u8]> {
         self.public_key.serialize()
     }
 
+    /// Deserialize a public identity from a byte slice.
     pub fn decode(value: &[u8]) -> Result<Self> {
         let pk = PublicKey::try_from(value)?;
         Ok(Self { public_key: pk })
     }
 
+    /// Given a trusted identity `self`, verify that `other` represents the same user for the given
+    /// `signature`.
```

Comment:
I'm not sure what you mean by "for the given signature". The signature is used to perform the verification, but if the verification passes, `self` and `other` represent the same user in all cases, not just regarding this signature.

Reply:
I was trying to clarify why the `signature` is needed for this method, and wasn't quite understanding what "alternate" identity was referring to. I have modified this to:
```rust
/// Given a trusted identity `self`, verify that `other` represents an alternate identity for
/// this user.
///
/// `signature` must be calculated from [`IdentityKeyPair::sign_alternate_identity`].
```

It looks like the alternate identity calculation mechanism is consumed in the app somewhere; I'll take a look at documenting that more fully in another PR perhaps.

Reply:
It's part of the Phone Number Privacy project but I don't know if anyone's hooked it up end-to-end on the app side yet. But as usual that's just the current use case; the API's designed for future sorts of "alternate identities" as well.